Tools/clinic/clinic.py

```diff
@@ -1943,12 +1943,12 @@ def dump(self):
 return_converters = {}
 
 
-def write_file(filename, new_contents):
+def write_file(filename, new_contents, force=False):
```

Comment:
I feel tempted to add type hints here, since gh-104050 is on.

```suggestion
def write_file(filename: str,
               new_contents: str, *,
               force: bool = False
) -> tuple[str, tuple[str, str]]:
```

Reply:
cc. @AlexWaygood

Reply:
`list[str, str]` isn't a valid type hint, which reinforces the need to run a type checker as a non-required check in CI if we're going to add type hints ;)

Reply:
`list[str]` indicates "a list with an arbitrary number of elements, all of which are of type `str`".

`tuple` is a very special type in the Python typing system, in that it's the only type that can have a variable number of parameters supplied to it. `tuple[int, int]` is conceptually a distinct type from `tuple[int, int, int]` from the type checker's perspective. But a str-list of length 2 is conceptually the same type as a str-list of length 3, and it's the same for the vast majority of variable-length Python containers

Reply:
Let's just drop this until we've got a CI check :)